### PR TITLE
TST: stats: mark tests slow/xslow

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
NA

#### What does this implement/fix?
Marks some slow tests slow/xslow as appropriate.

#### Additional information
There are still many tests in `test_continuous_basic.py` and `test_fit.py` that should probably be marked XSLOW, but I want to check on a different computer before pulling the trigger.
`scipy/stats/tests/test_stats.py::TestMGCStat::test_pvalue_literature` is actually xslow, but I thought we needed to leave at least one test not marked XSLOW.

### CODE REVIEW

File: scipy/stats/tests/test_axis_nan_policy.py [hunk 1]
```diff
@@ -17,6 +18,9 @@
 from scipy._lib._util import AxisError
 
 
+SCIPY_XSLOW = int(os.environ.get('SCIPY_XSLOW', '0'))
```
Comment (mdhaber) [inline scipy/stats/tests/test_axis_nan_policy.py:21]: Easier to skip the few tests this way than rewriting the way the cases are collected.

File: scipy/stats/tests/test_multivariate.py [hunk 2]
```diff
@@ -2583,7 +2583,7 @@ def test_cdf_signs(self):
         cdf = multivariate_normal.cdf(b, mean, cov, df, lower_limit=a)
         assert_allclose(cdf, cdf[0]*expected_signs)
 
-    @pytest.mark.parametrize('dim', [1, 2, 5, 10])
+    @pytest.mark.parametrize('dim', [1, 2, 5])
```
Comment (mdhaber) [inline scipy/stats/tests/test_multivariate.py:2586]: 2.80s call

File: scipy/stats/tests/test_morestats.py [hunk 3]
```diff
@@ -423,7 +423,7 @@ def test_example1b(self):
                                   tm[0:5], 4)
         assert_allclose(p, 0.0020, atol=0.00025)
 
-    @pytest.mark.slow
+    @pytest.mark.xslow
```
Comment (mdhaber) [inline scipy/stats/tests/test_morestats.py:426]: 2.55s call

File: scipy/stats/tests/test_fit.py [hunk 4]
```diff
@@ -901,7 +901,7 @@ def test_against_filliben_norm_table(self, case):
         res = stats.scoreatpercentile(res.null_distribution, percentiles*100)
         assert_allclose(res, ref, atol=2e-3)
 
-    @pytest.mark.slow
+    @pytest.mark.xslow
```
Comment (mdhaber) [inline scipy/stats/tests/test_fit.py:904]: ~3 s per case

File: scipy/stats/tests/test_fit.py [hunk 5]
```diff
@@ -840,7 +840,7 @@ def test_against_anderson_case_3(self):
         assert_allclose(res.statistic, 0.559)  # See [1] Table 1B 1.2
         assert_allclose(res.pvalue, 0.15, atol=5e-3)
 
-    @pytest.mark.slow
+    @pytest.mark.xslow
```
Comment (mdhaber) [inline scipy/stats/tests/test_fit.py:843]: 5.77s call

File: scipy/stats/tests/test_fit.py [hunk 6]
```diff
@@ -292,7 +292,7 @@ def cases_test_fit_mse():
                       'wald', 'weibull_max', 'weibull_min', 'wrapcauchy'}
 
     # Please keep this list in alphabetical order...
-    xslow_basic_fit = {'beta', 'betaprime', 'burr', 'burr12',
+    xslow_basic_fit = {'argus', 'beta', 'betaprime', 'burr', 'burr12',
```
Comment (mdhaber) [inline scipy/stats/tests/test_fit.py:295]: 1.61s call

File: scipy/stats/tests/test_fit.py [hunk 7]
```diff
@@ -79,7 +79,7 @@
 # Don't run the fit test on these:
 skip_fit = [
     'erlang',  # Subclass of gamma, generates a warning.
-    'genhyperbolic',  # too slow
+    'genhyperbolic', 'norminvgauss',  # too slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_fit.py:82]: 11.22s call

File: scipy/stats/tests/test_fast_gen_inversion.py [hunk 8]
```diff
@@ -132,13 +132,15 @@ def test_u_error(distname, args):
     assert u_error <= 1e-10
 
 
+@pytest.mark.xslow
```
Comment (mdhaber) [inline scipy/stats/tests/test_fast_gen_inversion.py:135]: 1.06s call

File: scipy/stats/tests/test_distributions.py [hunk 9]
```diff
@@ -7636,7 +7639,7 @@ def test_cdf_against_r(self, r_case_result):
             res = stats.studentized_range.cdf(q, k, v)
         assert_allclose(res, r_res)
 
-    @pytest.mark.slow
+    @pytest.mark.xslow
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:7642]: 4.99s call

File: scipy/stats/tests/test_distributions.py [hunk 10]
```diff
@@ -7589,7 +7591,7 @@ def test_pdf_against_mp(self, case_result):
                         atol=src_case["expected_atol"],
                         rtol=src_case["expected_rtol"])
 
-    @pytest.mark.slow
+    @pytest.mark.xslow
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:7594]: ~3s per case

File: scipy/stats/tests/test_distributions.py [hunk 11]
```diff
@@ -1861,6 +1861,7 @@ def test_mean_gh18511(self):
         rm = n / M * N
         assert_allclose(hm, rm)
 
+    @pytest.mark.xslow
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:1864]: 2.42s call

File: scipy/stats/tests/test_axis_nan_policy.py [hunk 12]
```diff
@@ -244,6 +248,8 @@ def nan_policy_1d(hypotest, data1d, unpacker, *args, n_outputs=2,
 def test_axis_nan_policy_fast(hypotest, args, kwds, n_samples, n_outputs,
                               paired, unpacker, nan_policy, axis,
                               data_generator):
+    if hypotest in {stats.cramervonmises_2samp} and not SCIPY_XSLOW:
```
Comment (mdhaber) [inline scipy/stats/tests/test_axis_nan_policy.py:251]: ~4s per call

File: scipy/stats/tests/test_continuous_basic.py [hunk 13]
```diff
@@ -38,72 +38,34 @@
...
-                        + ['pearson3'])
-skip_fit_fix_test = {"MLE": skip_fit_fix_test_mle,
-                     "MM": slow_fit_test_mm + fail_fit_fix_test_mm}
+# Sets of tests to skip.
+# Entries sorted by speed (very slow to slow).
+# xslow took > 1s; slow took > 0.5s
+
+xslow_test_cont_basic = {'studentized_range', 'kstwo', 'ksone', 'vonmises', 'kappa4',
+                         'recipinvgauss', 'vonmises_line', 'gausshyper', 'rel_breitwigner',
+                         'norminvgauss'}
```
Comment (mdhaber) [inline scipy/stats/tests/test_continuous_basic.py:47]:
```suggestion
                         'recipinvgauss', 'vonmises_line', 'gausshyper',
                         'rel_breitwigner', 'norminvgauss'}
```

**Comment by mdhaber** [comment]
Main:
w/ slow tests: 12526 passed, 582 skipped, 135 xfailed in 739.24s (0:12:19)
w/out slow tests: 8819 passed, 4378 skipped, 46 xfailed in 225.37s (0:03:45)

PR:
w/ slow tests: 12697 passed, 781 skipped, 134 xfailed in 445.68s (0:07:25)
w/out slow tests: 9042 passed, 4525 skipped, 45 xfailed in 171.51s (0:02:51)

File: scipy/stats/tests/test_continuous_basic.py [hunk 14]
```diff
@@ -920,6 +915,7 @@ def test_broadcasting_in_moments_gh12192_regression():
     assert vals3.shape == expected3.shape
 
 
+@pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_continuous_basic.py:918]: 0.93s call

File: scipy/stats/tests/test_distributions.py [hunk 15]
```diff
@@ -220,6 +220,7 @@ def test_fit_MLE_comp_optimizer(self, rvs_loc, rvs_shape,
         _assert_less_or_close_loglike(stats.vonmises, data,
                                       stats.vonmises.nnlf, **kwds)
 
+    @pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:223]: 0.52s call

File: scipy/stats/tests/test_distributions.py [hunk 16]
```diff
@@ -5082,6 +5084,7 @@ def nolan_loc_scale_sample_data(self):
         )
         return data
 
+    @pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:5087]: ~1s per call

File: scipy/stats/tests/test_distributions.py [hunk 17]
```diff
@@ -5110,7 +5113,7 @@ def test_rvs(
         )
         assert p > 0.05
 
-    @pytest.mark.slow
+    @pytest.mark.xslow
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:5116]: 2~3 seconds per call

File: scipy/stats/tests/test_distributions.py [hunk 18]
```diff
@@ -5208,6 +5211,7 @@ def test_fit_loc_extrap(self):
         assert alpha2 > 1, f"Expected alpha > 1, got {alpha2}"
         assert loc2 > max(x2), f"Expected loc > {max(x2)}, got {loc2}"
 
+    @pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:5214]: 0.15/1.23s per call

File: scipy/stats/tests/test_distributions.py [hunk 19]
```diff
@@ -6025,7 +6029,7 @@ def test_uniform_fit(self):
         assert_raises(ValueError, stats.uniform.fit, x, floc=2.0)
         assert_raises(ValueError, stats.uniform.fit, x, fscale=5.0)
 
-    @pytest.mark.slow
+    @pytest.mark.xslow
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:6032]: MM takes 1.66s

File: scipy/stats/tests/test_distributions.py [hunk 20]
```diff
@@ -7549,13 +7553,14 @@ class TestStudentizedRange:
         (1, 10, np.inf, 0.000519869467083),
     ]
 
+    @pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:7556]: 1.10s call

File: scipy/stats/tests/test_distributions.py [hunk 21]
```diff
@@ -7606,6 +7611,7 @@ def test_moment_against_mp(self, case_result):
                         atol=src_case["expected_atol"],
                         rtol=src_case["expected_rtol"])
 
+    @pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:7614]: 0.92s call

File: scipy/stats/tests/test_distributions.py [hunk 22]
```diff
@@ -9543,6 +9549,7 @@ def pdf(E, M, Gamma):
         ref = pdf(x, rho*Gamma, Gamma)
         assert_allclose(res, ref, rtol=rtol)
 
+    @pytest.mark.xslow
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:9552]: 1.85s call

File: scipy/stats/tests/test_hypotests.py [hunk 23]
```diff
@@ -1353,6 +1353,7 @@ def test_exact_pvalue(self, statistic, m, n, pval):
         # The values are taken from Table 2, 3, 4 and 5
         assert_equal(_pval_cvm_2samp_exact(statistic, m, n), pval)
 
+    @pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_hypotests.py:1356]: 0.53s call

File: scipy/stats/tests/test_kdeoth.py [hunk 24]
```diff
@@ -70,7 +70,7 @@ def test_kde_1d_weighted():
                         (kdepdf*normpdf).sum()*intervall, decimal=2)
 
 
-@pytest.mark.slow
+@pytest.mark.xslow
```
Comment (mdhaber) [inline scipy/stats/tests/test_kdeoth.py:73]: 4.28s call

File: scipy/stats/tests/test_kdeoth.py [hunk 25]
```diff
@@ -110,7 +110,7 @@ def test_kde_2d():
                         (kdepdf*normpdf).sum()*(intervall**2), decimal=2)
 
 
-@pytest.mark.slow
+@pytest.mark.xslow
```
Comment (mdhaber) [inline scipy/stats/tests/test_kdeoth.py:113]: 4.28s call

File: scipy/stats/tests/test_morestats.py [hunk 26]
```diff
@@ -2066,6 +2066,7 @@ def optimizer(fun):
         assert np.all(bounds[0] < maxlog)
         assert np.all(maxlog < bounds[1])
 
+    @pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_morestats.py:2069]: 0.78s call

File: scipy/stats/tests/test_multivariate.py [hunk 27]
```diff
@@ -2655,6 +2655,7 @@ def test_cdf_against_qsimvtv(self, dim, seed, singular):
             ref = _qsimvtv(20000, df, cov, a - mean, b - mean, rng)[0]
         assert_allclose(res, ref, atol=1e-4, rtol=1e-3)
 
+    @pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_multivariate.py:2658]: 1.12s call

File: scipy/stats/tests/test_odds_ratio.py [hunk 28]
```diff
@@ -105,6 +105,7 @@ def test_sample_odds_ratio_ci(self, case):
         ci = result.confidence_interval(confidence_level, alternative)
         assert_allclose([ci.low, ci.high], [ref_low, ref_high], rtol=1e-6)
 
+    @pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_odds_ratio.py:108]: 0.25~0.62s call

File: scipy/stats/tests/test_resampling.py [hunk 29]
```diff
@@ -161,6 +161,7 @@ def my_statistic(x, y, z, axis=-1):
     assert_equal(res2.standard_error.shape, result_shape)
 
 
+@pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_resampling.py:164]: 0.2~0.6s call

File: scipy/stats/tests/test_resampling.py [hunk 30]
```diff
@@ -304,6 +305,7 @@ def statistic(z, y, axis=0):
     assert_allclose(a_hat, 0.011008228344026734)
 
 
+@pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_resampling.py:308]: 0.68~1.08s call

File: scipy/stats/tests/test_resampling.py [hunk 31]
```diff
@@ -653,6 +656,7 @@ def statistic_1d(*data):
     assert_allclose(res1, res2)
 
 
+@pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_resampling.py:659]: ~0.7s per call

File: scipy/stats/tests/test_resampling.py [hunk 32]
```diff
@@ -847,6 +851,7 @@ def statistic(x, axis):
         assert_allclose(res.statistic, expected.statistic)
         assert_allclose(res.pvalue, expected.pvalue, atol=self.atol)
 
+    @pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_resampling.py:854]: ~0.4s per call

File: scipy/stats/tests/test_resampling.py [hunk 33]
```diff
@@ -928,6 +934,7 @@ def statistic1d(x):
...
```
Comment (mdhaber) [inline scipy/stats/tests/test_resampling.py:937]: 0.75s per call

File: scipy/stats/tests/test_resampling.py [hunk 34]
```diff
@@ -1088,6 +1095,7 @@ def test_input_validation(self):
         with pytest.raises(ValueError, match=message):
             power(test, rvs, n_observations, batch=10.5)
 
+    @pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_resampling.py:1098]: 1.08s call

File: scipy/stats/tests/test_resampling.py [hunk 35]
```diff
@@ -1123,6 +1131,7 @@ def test(x, axis):
         assert_equal(res1.power, res3.power)
         assert_equal(res2.power, res3.power)
 
+    @pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_resampling.py:1134]: 1.92s call; maybe should be xslow in the future

File: scipy/stats/tests/test_resampling.py [hunk 36]
```diff
@@ -1785,6 +1794,7 @@ def statistic1d(x, y):
                                >= np.abs(S-mean))/n
         assert_allclose(expected_Pr_gte_S_mean, Pr_gte_S_mean)
 
+    @pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_resampling.py:1797]: ~0.6s per call

File: scipy/stats/tests/test_stats.py [hunk 37]
```diff
@@ -5171,7 +5171,7 @@ def test_ttest_ind_permutation_check_p_values(self):
 class Test_ttest_ind_common:
     # for tests that are performed on variations of the t-test such as
     # permutations and trimming
-    @pytest.mark.slow()
+    @pytest.mark.xslow()
```
Comment (mdhaber) [inline scipy/stats/tests/test_stats.py:5174]: 1~2s per call

File: scipy/stats/tests/test_stats.py [hunk 38]
```diff
@@ -5676,6 +5676,7 @@ def test_ttest_single_observation():
     assert_allclose(res, (1.0394023007754, 0.407779907736), rtol=1e-10)
 
 
+@pytest.mark.slow
```
Comment (mdhaber) [inline scipy/stats/tests/test_stats.py:5679]: 0.61s call

File: scipy/stats/tests/test_stats.py [hunk 39]
```diff
@@ -7720,8 +7721,8 @@ def test_error_code(self):
             v_values = rng.random(size=(2, 2))
             _ = stats.wasserstein_distance_nd(u_values, v_values)
 
-    @pytest.mark.parametrize('u_size', [1, 10, 300])
```
Comment (mdhaber) [inline scipy/stats/tests/test_stats.py:7723]: 3.22s call

**Review by tupui — APPROVED** [review]
Nice gains and the marks look ok to me 👍

**Pull request merged**